internal/utils.go

```diff
@@ -280,3 +282,30 @@ func unmarshalManifestBytes(manifestBytes []byte) (*[]map[string]interface{}, er
 
 	return &yamlDocs, nil
 }
+
+// verifyManifestPath verifies that the manifest path is in the directory tree under baseDirectory.
+// An error is returned if it is not or the paths couldn't be properly resolved.
+func verifyManifestPath(baseDirectory string, manifestPath string) error {
+	absPath, err := filepath.Abs(manifestPath)
+	if err != nil {
+		return fmt.Errorf("could not resolve the manifest path %s to an absolute path", manifestPath)
+	}
+
+	relPath, err := filepath.Rel(baseDirectory, absPath)
+	if err != nil {
+		return fmt.Errorf(
+			"could not resolve the manifest path %s to a relative path from the kustomization.yaml file",
+			manifestPath,
+		)
+	}
+
+	parDir := ".." + string(filepath.Separator)
+	if strings.HasPrefix(relPath, parDir) || relPath == ".." {
```

Comment:
Do you even need `relPath`? Couldn't this just be:
```go
if strings.HasPrefix(baseDirectory, absPath)
```

Reply:
@dhaiducek in most cases, I think this would work but your suggestion would not work if the base directory is `/var/path/to/base` and `absPath` is `/var/path/to/base-camp/symlink-to-something`. It's very unlikely that we would encounter this in app subscription, but I think it's better to use the current solution in the PR.

What do you think?

Reply:
Ah, that's a good point. Yes, that is safer.